plugins/UFPWriter/UFPWriter.py

```diff
@@ -151,8 +156,30 @@ def _writeObjectList(archive):
 
         To retrieve, use: `archive.getMetadata(METADATA_OBJECTS_PATH)`
         """
+
         objects_model = CuraApplication.getInstance().getObjectsModel()
-        object_metas = [{"name": item["name"]} for item in objects_model.items]
+        object_metas = []
+
+        for item in objects_model.items:
+            object_metas = object_metas + UFPWriter._getObjectMetas(item["node"])
 
         data = {METADATA_OBJECTS_PATH: object_metas}
         archive.setMetadata(data)
+
+    @staticmethod
+    def _getObjectMetas(node: SceneNode) -> List[Dict]:
+        """Get object metadata to write for a Node.
+
+        :return: List of object metadata dictionaries.
+                 Might contain > 1 element in case of a group node.
+                 Might be empty in case of nonPrintingMesh
+        """
+
+        nodes = [node]
```

Comment:
Why not use a DepthFirstIterator for this?

Reply:
TIL